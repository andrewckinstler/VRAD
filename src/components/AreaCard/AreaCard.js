import React from 'react';
import './AreaCard.scss';

export const AreaCard = ({ nickname, longname, description }) => {
  return (
  <div>
    <h2>{nickname}</h2>
    <span>{longname}</span>
    <p>{description}</p>
    <button className='view-listings'>View listings</button>
  </div>
  )
}