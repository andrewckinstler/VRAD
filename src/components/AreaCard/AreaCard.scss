@mixin borders() {
  border: 1px solid #EBEBEB;
  border-radius: 4px;
  box-shadow: 0 2px 4px rgba(0,0,0,0.1);
}

.area-card {
  @include borders();
  text-align: center;
  margin: 15px;
  padding: 20px;
  width: 25%;

  h2 {
    margin-top: 25px;
  }

  span {
    display: inline-block;
    font-style: oblique;
    margin-bottom: 25px;
  }

  img {
    height: 10em;
    object-fit: cover;
    width: 15em;
  }

  button {
    @include borders();
    color: #0C4C38;
    font-size: .8em;
    font-weight: bold;
    height: 2.5em;
    margin: 20px;
    padding: 5px;
    width: 8em;

    &:hover {
      cursor: pointer;
      background-color: #28BB8D;
    }
  }
}
