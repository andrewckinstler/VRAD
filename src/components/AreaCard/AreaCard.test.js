import React from 'react';
import { shallow } from 'enzyme';
import { AreaCard } from './AreaCard';

describe('AreaCard', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    let mockArea = {
      nickname: 'test nickname',
      longname: 'test longname',
      description: 'test description',
      id: 5
    }
    const wrapper = shallow(<AreaCard
      {...mockArea} key={mockArea.id}
    />);
  expect(wrapper).toMatchSnapshot();
  });
});
