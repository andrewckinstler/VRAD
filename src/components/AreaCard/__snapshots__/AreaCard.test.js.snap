// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaCard should match the snapshot with all data passed in correctly 1`] = `
<div
  className="area-card"
>
  <h2>
    test nickname
  </h2>
  <span>
    test longname
  </span>
  <p>
    test description
  </p>
  <Link
    to="/area/5"
  >
    <button
      className="view-listings"
      onClick={[Function]}
    >
      View listings
    </button>
  </Link>
</div>
`;
